src/lazy/text/buffer.rs

```diff
@@ -1195,6 +1223,26 @@ mod tests {
         MatchTest::new(" ").expect_match(match_length(TextBufferView::match_stop_character));
     }
 
+    #[test]
+    fn test_match_ivm() {
+        fn match_ivm(input: &str) {
+            MatchTest::new(input).expect_match(match_length(TextBufferView::match_ivm));
+        }
+        fn mismatch_ivm(input: &str) {
+            MatchTest::new(input).expect_mismatch(match_length(TextBufferView::match_ivm));
+        }
+
+        match_ivm("$ion_1_0");
+        match_ivm("$ion_1_1");
+        match_ivm("$ion_1_2");
+        match_ivm("$ion_124_17");
+
+        mismatch_ivm("ion_1_0");
+        mismatch_ivm("$ion__1_0");
```

Comment:
Here's another test case to consider.
```suggestion
        mismatch_ivm("$ion__1_0");
        mismatch_ivm("$ion_1_0_0");
```